src/app/pages/quizzes-catalog/quizzes-catalog.component.ts

```diff
@@ -1,25 +1,43 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { UIKitModule } from '../../ui-kit/ui-kit.module';
-import { quizzesCatalog } from '../../test-data/quizzes-catalog';
 import { CommonModule } from '@angular/common';
 import {
   CardColors,
   CardItemStyle,
   cardItemStyles,
 } from '../../ui-kit/constants/card-item';
+import { CategoryModel } from '../../services/model/category.model';
+import { Subscription } from 'rxjs';
+import { CategoryService } from '../../services/catalog.service';
 
 @Component({
   standalone: true,
   selector: 'app-quizzes-catalog',
   templateUrl: './quizzes-catalog.component.html',
   imports: [ UIKitModule, CommonModule ],
 })
-export class QuizzesCatalogComponent {
-  quizzesCatalog = quizzesCatalog;
+export class QuizzesCatalogComponent implements OnInit, OnDestroy {
+  quizzesCatalog: CategoryModel[] = [];
+  subscription: Subscription = new Subscription();
+
+  constructor(private categoriesService: CategoryService) {}
+
+  ngOnInit(): void {
+    this.subscription = this.categoriesService.get().subscribe(
+      data => this.quizzesCatalog = data,
```

Comment:
let's use here catchError also, and display error message to the user, instead of alert.
You can try like this:
```
this.categoriesService.get().pipe(
tap((data) => this.quizzesCatalog = data),
catchError((error) => this.errorMessage = 'Here message for a user about the situation' or you can update message in categoriesService.get() to make it more user friendly and use it here as error.message
). subscribe();
```

Good article about catching errors: https://blog.angular-university.io/rxjs-error-handling/

Reply:
And I beleve we need to think about how do we want to cover the pending data state (variable, spinner or skeleton)